flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/savepoints/SavepointHandlers.java

```diff
@@ -267,29 +278,48 @@ public SavepointStatusHandler(
-                            (operationResult) -> {
-                                switch (operationResult.getStatus()) {
-                                    case SUCCESS:
-                                        return AsynchronousOperationResult.completed(
-                                                operationResultResponse(
-                                                        operationResult.getResult()));
-                                    case FAILURE:
-                                        return AsynchronousOperationResult.completed(
-                                                exceptionalOperationResultResponse(
-                                                        operationResult.getThrowable()));
-                                    case IN_PROGRESS:
-                                        return AsynchronousOperationResult.inProgress();
-                                    default:
-                                        throw new IllegalStateException(
-                                                "No handler for operation status "
-                                                        + operationResult.getStatus()
-                                                        + ", encountered for key "
-                                                        + key);
+                    .handle(
+                            (operationResult, throwable) -> {
+                                if (throwable == null) {
+                                    switch (operationResult.getStatus()) {
+                                        case SUCCESS:
+                                            return AsynchronousOperationResult.completed(
+                                                    operationResultResponse(
+                                                            operationResult.getResult()));
+                                        case FAILURE:
+                                            return AsynchronousOperationResult.completed(
+                                                    exceptionalOperationResultResponse(
+                                                            operationResult.getThrowable()));
+                                        case IN_PROGRESS:
+                                            return AsynchronousOperationResult.inProgress();
+                                        default:
+                                            throw new IllegalStateException(
+                                                    "No handler for operation status "
+                                                            + operationResult.getStatus()
+                                                            + ", encountered for key "
+                                                            + key);
+                                    }
+                                } else {
+                                    maybeHandleUnknownOperation(throwable, key);
```

Comment:
I don't like the method name. It does not indicate that an exception might be thrown here as well. As an alternative, we could generate an `Optional` of the `CompletionException` that is returned by the method and thrown in the else branch. That would make it more obvious.

Or we remove the method completely since it's only called at this location. But I feat that this might decrease the readability.

Reply:
> Or we remove the method completely since it's only called at this location. But I feat that this might decrease the readability.

Thinking about it once more: Should we handle UnknownOperations in `SavepointHandlerBase#triggerOperation` 🤔

Reply:
Why would we handle it in SavepointHandlerBase#triggerOperation? How could it return an UnknownOperationKeyException, when the very purpose of it is to trigger new operations?

Reply:
True, I didn't check that `SavepointHandlerBase` is only used as a parent for the savepoint creation handlers. What you're saying makes sense...